src/style.rs

```diff
@@ -0,0 +1,768 @@
+// SPDX-License-Identifier: MIT
+//
+// Copyright 2016-2025, Johann Tuffe.
+
+use std::fmt;
+
+/// Represents a color in ARGB format
+#[derive(Debug, Clone, Copy, PartialEq, Default)]
+pub struct Color {
+    /// Alpha channel (0-255)
+    pub alpha: u8,
+    /// Red channel (0-255)
+    pub red: u8,
+    /// Green channel (0-255)
+    pub green: u8,
+    /// Blue channel (0-255)
+    pub blue: u8,
+}
+
+impl Color {
+    /// Create a new color from ARGB values
+    pub fn new(alpha: u8, red: u8, green: u8, blue: u8) -> Self {
+        Self {
+            alpha,
+            red,
+            green,
+            blue,
+        }
+    }
+
+    /// Create a color from RGB values (alpha = 255)
+    pub fn rgb(red: u8, green: u8, blue: u8) -> Self {
+        Self::new(255, red, green, blue)
+    }
+
+    /// Create a color from ARGB integer
+    pub fn from_argb(argb: u32) -> Self {
+        Self {
+            alpha: ((argb >> 24) & 0xFF) as u8,
+            red: ((argb >> 16) & 0xFF) as u8,
+            green: ((argb >> 8) & 0xFF) as u8,
+            blue: (argb & 0xFF) as u8,
+        }
+    }
+
+    /// Convert to ARGB integer
+    pub fn to_argb(&self) -> u32 {
+        ((self.alpha as u32) << 24)
+            | ((self.red as u32) << 16)
+            | ((self.green as u32) << 8)
+            | (self.blue as u32)
+    }
+
+    /// Check if the color is transparent
+    pub fn is_transparent(&self) -> bool {
```

Comment:
Transparency is usually stored as a separate parameter outside the color. I don't think this would actually tell the user if the color was transparent.